core/src/test/java/org/kiwix/kiwixmobile/core/page/bookmark/viewmodel/BookmarkViewModelTest.kt

```diff
@@ -73,18 +69,13 @@ internal class BookmarkViewModelTest {
     every { zimReaderContainer.name } returns "zimName"
     every { sharedPreferenceUtil.showBookmarksAllBooks } returns true
     every { bookmarksDao.bookmarks() } returns itemsFromDb.distinctUntilChanged()
+    every { bookmarksDao.pages() } returns bookmarksDao.bookmarks()
     viewModel = BookmarkViewModel(bookmarksDao, zimReaderContainer, sharedPreferenceUtil)
   }
 
   @Test
-  fun `initial state is Initialising`() {
-    viewModel.state.test().assertValue(bookmarkState())
-  }
-
-  @Test
-  internal fun `ExitBookmarks finishes activity`() {
-    viewModel.effects.test().also { viewModel.actions.offer(Exit) }.assertValue(Finish)
-    viewModel.state.test().assertValue(bookmarkState())
+  fun `Initial state returns initial state`() {
```

Comment:
```
 override fun updatePagesBasedOnFilter(
    state: BookmarkState,
    action: Action.Filter
  ): BookmarkState =
    state.copy(searchTerm = action.searchTerm)

  override fun updatePages(
    state: BookmarkState,
    action: Action.UpdatePages
  ): BookmarkState =
    state.copy(pageItems = action.pages.filterIsInstance<BookmarkItem>())

  override fun offerUpdateToShowAllToggle(
    action: Action.UserClickedShowAllToggle,
    state: BookmarkState
  ): BookmarkState {
    effects.offer(UpdateAllBookmarksPreference(sharedPreferenceUtil, action.isChecked))
    return state.copy(showAll = action.isChecked)
  }

  override fun deselectAllPages(state: BookmarkState): BookmarkState =
    state.copy(pageItems = state.pageItems.map { it.copy(isSelected = false) })

  override fun createDeletePageDialogEffect(state: BookmarkState) =
    ShowDeleteBookmarksDialog(effects, state, pageDao)

  override fun copyWithNewItems(state: BookmarkState, newItems: List<BookmarkItem>): BookmarkState =
    state.copy(pageItems = newItems)
```

I want these methods tested similarly to `initalState()` and probably the deletion of a lot of these other tests